### PR TITLE
Add resource limits for upgrade and delete hook batch jobs

### PR DESCRIPTION
**Pull request by kfox1111**
Updated pr: https://github.com/spiffe/helm-charts-hardened/pull/322 here since it is unwritable.

### CODE REVIEW

**Review by edwbuck — CHANGES_REQUESTED** [review]
Requesting clarity on the desired "resource" comments and "commenting pattern"

This is because the pattern itself doesn't align with our desires, as we don't want people editing this file, and the helm chart itself doesn't need an object defined here (because we are using better practices than were there originally).

File: charts/spire/values.yaml [hunk 1]
```diff
@@ -89,9 +89,33 @@ global:
...
+      #   cpu: 100m
+      #   memory: 128Mi
+      # requests:
+      #   cpu: 100m
+      #   memory: 128Mi
   deleteHooks:
     ## @param global.deleteHooks.enabled Enable Helm hooks to autofix common delete issues (should be disabled when using `helm template`)
     enabled: true
+    ## @param global.deleteHooks.resources [object] Resource requests and limits for deleteHooks
+    resources: {}
```
[the remark this answers is not in the document]
Comment (kfox1111) [reply charts/spire/values.yaml:108]: updated as discussed in the meeting yesterday.

**Review by edwbuck — APPROVED** [review]
Apparently the desire is to keep the resources pattern as-is.  This is deduced by the comment "updated in accordance to the meeting (of June 18th)" where the meeting held the door open for any desired improvements.

Approving

**Comment by edwbuck** [comment]
@kfox1111 There are failures in the checks.  It looks like some logic is cutting off the first letter of the name of the host in one test suite.  Maybe that's because it assumed a https:// prefix somewhere, and the prefix is http://?  Please look into it and see if this can be fixed prior to merge.

**Comment by kfox1111** [comment]
There s a race that happens occasionally. Retesting the failed job in that case resolves it. There is an open issue to fix it.

**Pull request merged**